### ISSUE TITLE
Comments on the single line localization

### ISSUE THREAD

**Issue by abrarum** [abrarum/bezierobjdet#3]
Hi Alexander. I've updated the git with an [index file](https://github.com/abrarum/bezierobjdet/blob/main/TESTS/workspace/mobile/index.ipynb). It has a pretty low loss value after 50 epochs (although I've trained it a lot, might be overfitted by now b.c of an error in the metric but will update that). Please let me know with your feedback and comments? 

I now want to achieve multi-line detection where I will now add "curves" (more than 2 control points) as a second label with the existing straight-line label in my dataset. What do you think?

**Comment by alexander-g** [comment]
Hi Abrar, looking good! :rocket: 
Yes I also think it might be overfitted, would be interesting to see some test images.

I will try to run it later, so far I've only noticed that your training looks a bit slow. This might be due to using a generator as data source. You could try using the arguments `use_multiprocessing=True` and `workers=X` (with X>1) in `.fit()`. This might duplicate some images but this should not matter. Alternatively use `tf.data.Dataset` with `.map(...,-1)`

I suggest to keep the 2 control points for the moment, going from a single line to multi-line will be already quite a big step. What's your plan how you would implement this?

If it was me, I would:
- Replace the Dense layers with convolutions so that you get a `NxN` grid as output
- For each image, create a `NxNx1` target for the classification head, with `1` if the grid cell is close to/contains a line, `0` otherwise. Finding the best condition for 1 or 0 is part of the topic
- For each image, create a `NxNx4` target for the regression head, encoding control points of the closest line relative to the grid cell center. Only train on cells where classification target is `1`, ignore on `0`.
- Maybe think about non-maximum suppression at the end

This is basically what a box-based object detector does.

**Comment by abrarum** [comment]
I've seen your message already. I'll be on to it. I haven't figured out the multi-line detection yet so no idea about the approach but I'll be moving with your recommendations..
